railties/lib/rails/generators/generated_attribute.rb

```diff
@@ -185,7 +185,14 @@ def polymorphic?
       end
 
       def required?
-        reference? && Rails.application.config.active_record.belongs_to_required_by_default
+        if required_by_default = Rails.application.config.active_record.belongs_to_required_by_default
```

Comment:
Maybe I'm missing something, but outside of its usage here for generating migrations, I don't see the global config being used when building the association 🤔